app/models/miq_worker/runner.rb

```diff
@@ -427,6 +427,28 @@ def safe_sleep(seconds)
     sleep(seconds % SAFE_SLEEP_SECONDS)
   end
 
+  def self.ruby_object_usage
+    types = Hash.new { |h, k| h[k] = 0 }
+    ObjectSpace.each_object do |obj|
+      types[obj.class.name] += 1
+    end
+    types
+  end
+
+  LOG_RUBY_OBJECT_USAGE_INTERVAL = 60
+  def self.log_ruby_object_usage(top = 20)
+    return unless top > 0
+
+    t = Time.now.utc
+    @last_ruby_object_usage ||= t
+
+    if (@last_ruby_object_usage + LOG_RUBY_OBJECT_USAGE_INTERVAL) < t
+      types = ruby_object_usage
+      _log.info("Ruby Object Usage: #{types.sort_by { |_k, v| -v }.take(top).inspect}")
```

Comment:
@Fryguy Took your suggestion and simplified the ruby_object_usage hash and the logging here.

Also, I made these previously protected/mostly unused methods public class methods so you can now call `MiqWorker::Runner. log_ruby_object_usage`.  These could probably go in a utility class but wanted to clean these up specifically so I could drop the poll value for worker.